clients/shared/config/apiConfig.ts

```diff
@@ -0,0 +1,24 @@
+import { fetchAuthSession } from '@aws-amplify/auth'
+import axios from 'axios'
+
+const api = axios.create({
+  baseURL: import.meta.env.VITE_API_URL,
+  headers: {
+    'Content-Type': 'application/json'
+  }
+})
+
+api.interceptors.request.use(
+  async(config) => {
+    const idToken = (
+      await fetchAuthSession()
+    ).tokens?.idToken?.toString()
+    if (idToken !== undefined && idToken !== null) {
```

Comment:
should this be `!idToken`

Reply:
we have added `@typescript-eslint/restrict-template-expressions` in the `eslint` config. it seems more intuitive to me to follow an explicit expression that is more generic and similar to other languages rather than js nuisance. if the team feels like, of course, they can choose otherwise and remove this rule.